### PR TITLE
Added Sync Fork workflow

### PR DESCRIPTION
**Pull request by Bobbins228**
## Description

A workflow that will automatically sync the opendatahub-io/codeflare-operator fork
relates to dw [#13](https://github.com/red-hat-data-services/distributed-workloads/issues/13) & cfo[#327](https://github.com/project-codeflare/codeflare-operator/issues/327)
## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: .github/workflows/sync-fork.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      GIT_BRANCH: ${GITHUB_REF#refs/heads/}
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        persist-credentials: false
+    - name: repo-sync
+      uses: repo-sync/github-sync@v2
+      with:
+        source_repo: "https://github.com/opendatahub-io/codeflare-operator.git"
```
Comment (VanillaSpoon) [inline .github/workflows/sync-fork.yaml:22]: I'm not certain if this is the correct [Repo for github-sync](https://github.com/repo-sync/github-sync/blob/master/github-sync.sh), and I might be reading it wrong, but should the source_repo be from project-codeflare/codeflare-operator?

Comment (dimakis) [reply .github/workflows/sync-fork.yaml:22]: great catch @VanillaSpoon 👍

**Pull request merged**